src/main.py

```diff
@@ -28,7 +28,69 @@
 IO.setup(15,IO.IN)
 
 def main():
-    return 
+        #Initilize
+
+    COLD_VALVE_SERVO = 0
+    HOT_VALVE_SERVO = 1
+    kit = ServoKit(channels=16)
+
+    # ensure water valves are off, soap line off and water directed towards fauct 
+    set_cold_valve_angle(0)
+    set_hot_valve_angle(0)
+    three_way(False)
+    two_way(False)
+
+    while True:
+        # IO.input(x) returns true if path is clear
+
+        # either IR sensor detects a hand, run hand washer
+        while not IO.input(11) or not IO.input(13):
+
+            # direct 3-way valve to hand wash line
+            three_way(True)
+            print("directed towards hand washer")
+
+            # turn on water
+            # run for a bit to get hands wet
+            water_on()
+            print("first rinse")
+            time.sleep(2)
+
+            # open soap valve
+            # cycle and then end with water
+            two_way(True)
+            print("first soap cycle")
+            time.sleep(2)
+            two_way(False)
+            time.sleep(2)
+            two_way(True)
+            print("second soap cycle")
+            time.sleep(2)
+            two_way(False)
+
+            # final rinse
+            print("final rinse")
+            time.sleep(5)
+
+            # cut water
+            water(False)
+
+        # hand removed    
+        else:
+            print("Sensor 1 and 2 deactivated")
+            # ensure everything is off
+            two_way(False)
+            three_way(False)
+            water(False)
+
+        # run regular faucet
+        while not IO.input(15):
+            print("Sensor 3 active")
+            three_way(False)
+            water_on(True)
+        else:
+            print("Sensor 3 deactivated")
+            water_on(False) 
```

Comment:
Same here this will always run. We don't really want that since it will constantly draw power. 

We could save the "state" of the unit using a toggle. For example - `isOff` and only run the code in the else statement when this toggle is on. We can switch the toggle on and off when the state changes.